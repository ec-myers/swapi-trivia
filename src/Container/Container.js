import React from 'react';
import MovieCard from '../MovieCard/MovieCard';
import CharacterCard from '../CharacterCard/CharacterCard';

const Container = ({cards, goToMovieCharacters}) => {
  let allCards = cards.map(card => {
    if (card.title) {
      return <MovieCard key={card.episode} movie={card} goToMovieCharacters={goToMovieCharacters}/>
    }
    return <CharacterCard key={card.name} character={card} />
  })

  return(<section>
    {allCards}
  </section>)
}

export default Container;