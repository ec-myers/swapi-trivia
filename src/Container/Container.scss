$accent-color-dark: rgba(108, 91, 117, 0.8);
$accent-color-light: rgba(242,136,79, 0.8);

.Container {
    padding: 35px;
    display: grid;
    grid-template-columns: repeat(3, 1fr);
    grid-gap: 20px;
    overflow: scroll;
    height: 100vh;
}

.empty-favorites {
  display: flex;
  flex-direction: column;
  color: white;
  align-items: center;
  height: 100vh;
  padding-top: 20px;
  h3 {
    font-size: 3em;
  }
  p {
    font-size: 2em;
  }
}