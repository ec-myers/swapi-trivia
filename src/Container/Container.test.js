import React from 'react'
import {shallow} from 'enzyme';
import Container from './Container'

describe('Container with Movies', () => {
  it('should match the snapshot when rendering movies', () => {
    let movies =[
      {
        title: 'Bad Prequel',
        episode: 1,
        releaseYear: 2001,
        id: 3
      },
      {
        title: 'Worse Prequel',
        episode: 2,
        releaseYear: 2007,
        id: 7
      }
    ]
    let wrapper = shallow(<Container
      cards={movies} goToMovies={jest.fn()}
    />)
    expect(wrapper).toMatchSnapshot()
  })
})

describe('Container with Characters', () => {
  it('should match the snapshot when rendering characters', () => {
    let characters = [{
      name: 'Luke Skywalker',
      homeworld: 'Tattooine',
      population: 200000,
      species: 'human',
      films: ['blah', 'blah']
    }]
    let wrapper = shallow(<Container
      type='characters' cards={characters} toggleFavorite={jest.fn()}
    />)
    expect(wrapper).toMatchSnapshot()
  })
})

describe('Container with no favorites', () => {
  it('should match the snapshot when rendering and favorites is empty', () => {
    let favorites = []
    let wrapper = shallow(<Container
      type='favorites' cards={favorites}
      favorites={favorites}
      toggleFavorite={jest.fn()}
    />)
    expect(wrapper).toMatchSnapshot()
  })
})

describe('Container with Favorites', () => {
  it('should match the snapshot when rendering multiple favorites', () => {
    let favorites = [{
      name: 'Luke Skywalker',
      homeworld: 'Tattooine',
      population: 200000,
      species: 'human',
      films: ['blah', 'blah']
    }, {
        name: 'Luke C3-PO',
        homeworld: 'Tattooine',
        population: 200000,
        species: 'Droid',
        films: ['All of them', 'blah']
    }]
    let wrapper = shallow(<Container
      type='favorites' cards={favorites}
      favorites={favorites} toggleFavorite={jest.fn()}
    />)
    expect(wrapper).toMatchSnapshot()
  })
})