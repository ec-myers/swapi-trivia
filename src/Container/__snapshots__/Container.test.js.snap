// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container with Characters should match the snapshot when rendering characters 1`] = `
<section
  className="Container"
>
  <CharacterCard
    character={
      Object {
        "films": Array [
          "blah",
          "blah",
        ],
        "homeworld": "Tattooine",
        "name": "Luke Skywalker",
        "population": 200000,
        "species": "human",
      }
    }
    key="Luke Skywalker"
    toggleFavorite={[MockFunction]}
  />
</section>
`;

exports[`Container with Favorites should match the snapshot when rendering multiple favorites 1`] = `
<section
  className="Container"
>
  <CharacterCard
    character={
      Object {
        "films": Array [
          "blah",
          "blah",
        ],
        "homeworld": "Tattooine",
        "name": "Luke Skywalker",
        "population": 200000,
        "species": "human",
      }
    }
    favorites={
      Array [
        Object {
          "films": Array [
            "blah",
            "blah",
          ],
          "homeworld": "Tattooine",
          "name": "Luke Skywalker",
          "population": 200000,
          "species": "human",
        },
        Object {
          "films": Array [
            "All of them",
            "blah",
          ],
          "homeworld": "Tattooine",
          "name": "Luke C3-PO",
          "population": 200000,
          "species": "Droid",
        },
      ]
    }
    key="Luke Skywalker"
    toggleFavorite={[MockFunction]}
  />
  <CharacterCard
    character={
      Object {
        "films": Array [
          "All of them",
          "blah",
        ],
        "homeworld": "Tattooine",
        "name": "Luke C3-PO",
        "population": 200000,
        "species": "Droid",
      }
    }
    favorites={
      Array [
        Object {
          "films": Array [
            "blah",
            "blah",
          ],
          "homeworld": "Tattooine",
          "name": "Luke Skywalker",
          "population": 200000,
          "species": "human",
        },
        Object {
          "films": Array [
            "All of them",
            "blah",
          ],
          "homeworld": "Tattooine",
          "name": "Luke C3-PO",
          "population": 200000,
          "species": "Droid",
        },
      ]
    }
    key="Luke C3-PO"
    toggleFavorite={[MockFunction]}
  />
</section>
`;

exports[`Container with Movies should match the snapshot when rendering movies 1`] = `
<section
  className="Container"
>
  <MovieCard
    key="1"
    movie={
      Object {
        "episode": 1,
        "id": 3,
        "releaseYear": 2001,
        "title": "Bad Prequel",
      }
    }
  />
  <MovieCard
    key="2"
    movie={
      Object {
        "episode": 2,
        "id": 7,
        "releaseYear": 2007,
        "title": "Worse Prequel",
      }
    }
  />
</section>
`;

exports[`Container with no favorites should match the snapshot when rendering and favorites is empty 1`] = `
<div
  className="empty-favorites"
>
  <h3>
    Add some favorites!
  </h3>
  <p>
    Head back to the 'Movies' Tab
  </p>
</div>
`;
